From 8d13e09ae2b22ffcdfa784befb6b8af8a6da5b71 Mon Sep 17 00:00:00 2001
From: Robert Richardson <robert.richardson@suse.com>
Date: Mon, 11 Apr 2022 12:27:09 +0200
Subject: [PATCH] test patch 01

---
 test/TEST-01-BASIC/test.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/test/TEST-01-BASIC/test.sh b/test/TEST-01-BASIC/test.sh
index 012f2e5a..8c68fe40 100755
--- a/test/TEST-01-BASIC/test.sh
+++ b/test/TEST-01-BASIC/test.sh
@@ -39,13 +39,13 @@ test_setup() {
             mkdir -p root usr/bin usr/lib usr/lib64 usr/sbin
         )
         inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip \
-            mount dmesg dhclient mkdir cp ping dhclient \
-            umount strace less setsid dd sync
+            mount dmesg mkdir cp ping \
+            umount less setsid dd sync qemu
+        inst_multiple -o wicked dhclient arping arping2
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
             [ -f ${_terminfodir}/l/linux ] && break
         done
         inst_multiple -o ${_terminfodir}/l/linux
-        inst "$basedir/modules.d/35network-legacy/dhclient-script.sh" "/sbin/dhclient-script"
         inst "$basedir/modules.d/35network-legacy/ifup.sh" "/sbin/ifup"
 
         inst_simple "${basedir}/modules.d/99base/dracut-lib.sh" "/lib/dracut-lib.sh"
-- 
2.35.1

